CI should test whether generated files match latest sources

The CI should test whether the generated files (`expressions.go` and `y.go`) match the sources.

The following GitHub Action will do this:

```yaml
  - name: Test generated code
  - run: |
      make generate
      git diff --exit-code
```

The missing pieces are:

## 1. Installing Ragel

This works:

```yaml
  - name: Install Ragel
    run: |
      sudo apt-get update
      sudo apt-get install 
```

However, this requires `apt-get update` and `apt-get install` on each run.

Is there a way to cache this? Alternatively, should this action use a Docker image that includes Ragel?

## 2. Installing goyacc

I tried this:

Add a step:

```yaml
  - name: Install generators
    run: make setup
  - name: Run generators
    run: make generate
```

(`make setup` includes `go install golang.org/x/tools/cmd/goyacc`. `make generate` runs `go generate ./...`.)

This produces an error:

```sh
expressions/parser.go:3: running "goyacc": exec: "goyacc": executable file not found in $PATH
make: *** [Makefile:24: generate] Error 1
Error: Process completed with exit code 2.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -73,6 +73,35 @@ jobs:
           exit 1
         fi
 
+  verify-generated:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Install Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: 1.24.x
+
+    - name: Install tools
+      run: make tools
+
+    - name: Regenerate yacc parser
+      run: |
+        cd expressions
+        goyacc expressions.y
+        gofmt -w y.go
+
+    - name: Check for differences
+      run: |
+        if [ -n "$(git status --porcelain)" ]; then
+          echo "Generated parser files are out of date. Please run 'go generate ./...' and commit the results."
+          git diff
+          exit 1
+        fi
+
   verify-mod:
     runs-on: ubuntu-latest
 

